packages/eslint-plugin-internal/src/rules/react-typescript-prettier.js

```diff
@@ -10,4 +10,10 @@ module.exports = {
       unnamedComponents: 'arrow-function',
     },
   ],
+
+  // Allow props spreading
+  'react/jsx-props-no-spreading': 'off',
+
+  // Disable default props require
+  'react/require-default-props': 'off',
```

Comment:
@fhlavac I really think we should keep these rules enabled if possible. 

`jsx-props-no-spreading` - while I like this for easy component development, we need to think of our components as an API, and tend to agree with the eslint doc:
> This enhances readability of code by being more explicit about what props are received by the component. It is also good for maintainability by avoiding passing unintentional extra props and allowing react to emit warnings when invalid HTML props are passed to HTML elements

`require-default-props` - I think this forces us to consider which props are required and which are not WRT our component API.

> One advantage of defaultProps over custom default logic in your code is that defaultProps are resolved by React before the PropTypes typechecking happens, so typechecking will also apply to your defaultProps.

Reply:
Completely agree with the `require-default-props` just not sure about `jsx-props-no-spreading ` since in https://github.com/openshift/dynamic-plugin-sdk/pull/34/files#diff-b56b27c61ebe92f1fe8626d77d6dcae22ddfbccfd4c1d71f7da9d3c629e615c5R101 we probably want to spread any properties passed. Disabled the rule for that line for now.